source/extensions/common/aws/utility.cc

```diff
@@ -25,19 +32,27 @@ Utility::canonicalizeHeaders(const Http::RequestHeaderMap& headers) {
         if (!entry.key().getStringView().empty() && entry.key().getStringView()[0] == ':') {
           return Http::HeaderMap::Iterate::Continue;
         }
+        // For S3, we restrict the headers even further to avoid  headers that might mutate
+        // when in transit (e.g.: x-forwarded-for, internal headers, etc).
+        if (*pair->second == "s3") {
```

Comment:
This doesn't apply just to s3. You can remove this check.

Reply:
True(-ish). Comparing https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html to https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html, it looks like S3 has a smaller set of minimally required headers -- no?

Also, S3 requests are more likely to see headers mutations given it's a typical use case to have regular HTTP traffic --> Envoy --> S3?